app/src/main/java/ch/epfl/sweng/swenggolf/statistics/UserStats.java

```diff
@@ -0,0 +1,170 @@
+        return stats.size();
+    }
+
+    /**
+     * Writes the stats of a specific User to the database.
+     * @param userId the id of the specific user
+     */
+    public void write(String userId) {
+        Log.d(LOG_KEY_STATISTICS, "write stats for user "+userId);
+        Database.getInstance()
+                .write(Database.STATISTICS_USERS_PATH, userId, this);
+    }
+
+    /**
+     * Reads the stats of a specific User from the database.
+     * @param listener the ValueListener associated to the operation
+     * @param userId the id of the User
+     */
+    public static void read(final ValueListener<UserStats> listener, String userId) {
+        Log.d(LOG_KEY_STATISTICS, "read stats for user "+userId);
+        Database.getInstance()
+                .read(Database.STATISTICS_USERS_PATH, userId, listener, UserStats.class);
+    }
+
+    /**
+     * Increments (by a specific amount) a statistic for a specific User.
+     * @param stat the specific Stat to increment
+     * @param userId the id of the user
+     * @param nb The amount to increment
+     */
+    public static void updateStat(final Stats stat, final String userId, final int nb){
+        ValueListener<UserStats> listener = new ValueListener<UserStats>() {
+            @Override
+            public void onDataChange(UserStats stats) {
+                Log.d(LOG_KEY_STATISTICS, "Updated "+stat.toString()+" for user " + userId);
+                int oldValue = stats.getValue(stat);
+                stats.put(stat, oldValue+nb);
+                stats.write(userId);
+            }
+
+            @Override
+            public void onCancelled(DbError error) {
+                checkBackwardsCompatibility(error, userId);
+            }
+        };
+
+        read(listener, userId);
+    }
+
+    /**
+     * Handles the case where it it an old User without statistics in the database yet.
+     * @param error the error returned by the database
+     * @param userId the id of the specific User
+     */
+    public static void checkBackwardsCompatibility(DbError error, String userId) {
+        if (error == DbError.DATA_DOES_NOT_EXIST){
+            UserStats initStats = new UserStats();
+            initStats.write(userId);
+            Log.d(LOG_KEY_STATISTICS, "Stats generated for user "+userId);
+        } else {
+            Log.e(LOG_KEY_STATISTICS, "Failed to load stats for user " + userId);
+        }
+    }
+
+    public enum Stats {
+        OFFERS_CREATED("Offers created"),
+        OFFERS_CLOSED("Offers closed"),
+        OFFERS_DELETED("Offers deleted"),
+        OFFERS_READ("Offers read"),
+        OFFERS_TOTAL_VIEWS("Total views of own offers"),
+        MESSAGES_SENT("Direct messages sent"),
+        MESSAGES_RECEIVED("Direct messages received"),
+        ANSWERS_POSTED("Answers posted"),
+        ANSWERS_ACCEPTED("Own answers accepted"),
+        LOGIN("Total logins");
+
+        private final String text;
+
+        Stats(String text){
```

Comment:
Another tiny detail, some of your functions could benefit from formatting (it's just a missing space, I know, I'm annoying)